### PR TITLE
feat: add bumpoverride option

### PR DESCRIPTION
**Pull request by MisterTimn**
# Pull Request

## Description

I use conventional commits for changelog generation mainly and have a need to manually set the bump level from time to time (custom CalVer based versioning scheme). Other tools like `standard-version` of npm also allow for override through a `--releaseAs=patch|minor|major` parameter.

This PR simply adds an optional `--bumpoverride=patch|minor|major` argument. If it's not set then automatic version bumping is used as before, when set the `rec_bump` is overriden.

```
~/forks/version-bumper   master  docker run -v $PWD:/usr/share/repo jveessen/version-bumper:latest                  
2.1.0 2.1.1.dev
~/forks/version-bumper   master ±  docker run -v $PWD:/usr/share/repo jveessen/version-bumper:latest --bumpoverride minor                 
2.1.0 2.1.1.dev
~/forks/version-bumper   master ±  docker run -v $PWD:/usr/share/repo jveessen/version-bumper:latest --bumpoverride patch                 
2.0.5 2.0.6.dev
~/forks/version-bumper   master ±  docker run -v $PWD:/usr/share/repo jveessen/version-bumper:latest --bumpoverride major                 
3.0.0 3.0.1.dev
```

## Checklist

- [ ] I have followed this repository's contributing guidelines.
- [x] I have used the conventional commits specification for my commit messages.
- [x] I have signed my commits.
- [x] I will adhere to the project's code of conduct.

## Additional information
The contributing guidelines link seems to be broken so I couldn't check them.

> Anything else?
>

### CODE REVIEW

File: entrypoint.sh
Comment (TomerFi) [inline entrypoint.sh:74]: can you remove this redundant extra line here, please?

File: entrypoint.sh
Comment (TomerFi) [inline entrypoint.sh:73]: I think the linting workflow doesn't like this no-default value.

You can lint locally to verify, find instructions in the [contributing guidelines](https://github.com/TomerFi/version-bumper/blob/master/.github/CONTRIBUTING.md#lint-sources)

File: entrypoint.sh
Comment (TomerFi) [inline entrypoint.sh:88]: why is this `exit` meant to do here?

**Review by TomerFi — CHANGES_REQUESTED** [review]
Hey @MisterTimn 

Thank you very much - this looks great!
I just have a couple of change requests and we can get this merged and published quickly.

:-)

File: entrypoint.sh
Comment (MisterTimn) [reply entrypoint.sh:88]: This was autofixed by vscode (through extension and shellcheck)
Relevant shellcheck entry: https://www.shellcheck.net/wiki/SC2164

> cd can fail for a variety of reasons: misspelled paths, missing directories, missing permissions, broken symlinks and more.
> If/when it does, the script will keep going and do all its operations in the wrong directory. This can be messy, especially if the operations involve creating or deleting a lot of files.
> To avoid this, make sure you handle the cases when cd fails. Ways to do this include
> cd foo || exit as suggested to abort immediately, reusing exit code from failed cd command
> cd foo || { echo "Failure"; exit 1; } abort with custom message
> cd foo || ! echo "Failure" omitting "abort with custom message"
> if cd foo; then echo "Ok"; else echo "Fail"; fi for custom handling
<(cd foo && cmd) as an alternative to <(cd foo || exit; cmd) in <(..), $(..) or ( )

File: entrypoint.sh
Comment (MisterTimn) [reply entrypoint.sh:73]:
```
2022-08-01 12:06:49 [INFO]   File:[/tmp/lint/entrypoint.sh]
2022-08-01 12:06:49 [INFO]    - File:[entrypoint.sh] was linted with [shfmt] successfully
```
Locally no linting errors are given 😕

File: entrypoint.sh
Comment (TomerFi) [reply entrypoint.sh:88]: yeah, but we don't want the app exiting at this point,
an error thrown will be much more useful.

you can [ignore](https://github.com/koalaman/shellcheck/wiki/Directive) this specific error,
add `# shellcheck disable=SC2164` above that statement.

Comment (TomerFi) [reply entrypoint.sh:87]: BTW
is the `shellcheck` extension the one responsible for adding all of these quotation marks?

because I seem to remember an issue I had with them, but I don't remember what it was.

anyway, we use [shfmt](https://github.com/mvdan/sh) for linthing shell scripts for this repository, so maybe it's best if you drop the `shellcheck` vscode extension for this specific project and revert back all the modifications it made?
it looks like it doesn't see eye-to-eye with `shfmt`.

File: entrypoint.sh
Comment (TomerFi) [reply entrypoint.sh:73]: yes, my bad the linting workflow was screaming about the following redundant empty line and not this statement:

image

**Comment by TomerFi** [comment]
@MisterTimn 
Thank you very much for your contribution,
I'll merge this now and release a new version right asap.

About these failing CI workflows, you have 3 commits in this PR,
one of them isn't following the conventional commits specifications,
another one is not signed off, and another is a good commit.

As this is gonna be squashed before merging and we have one good commit, we can ignore these:

image

One more point I have,
for future PRs you might prefer to create a branch on your fork before you open a PR.
When you PR from the default branch:
1- CLI users will have to replace references if they want to pitch in on your project, including maintainers that might want to help.
2- As soon as this gets merged - your fork is gonna get out of sync, and you might need some manual commands to get it right. (or recreate the fork).

Thank you very much - merging now!

**Comment by MisterTimn** [comment]
> @MisterTimn Thank you very much for your contribution, I'll merge this now and release a new version right asap.
> 
> About these failing CI workflows, you have 3 commits in this PR, one of them isn't following the conventional commits specifications, another one is not signed off, and another is a good commit.
> 
> As this is gonna be squashed before merging and we have one good commit, we can ignore these:
> 
> image
> 
> One more point I have, for future PRs you might prefer to create a branch on your fork before you open a PR. When you PR from the default branch: 1- CLI users will have to replace references if they want to pitch in on your project, including maintainers that might want to help. 2- As soon as this gets merged - your fork is gonna get out of sync, and you might need some manual commands to get it right. (or recreate the fork).
> 
> Thank you very much - merging now!

Thanks for the info and future tips :)

**Comment by TomerFi** [comment]
@all-contributors please add @MisterTimn for code

**Comment by TomerFi** [comment]
@all-contributors please add @MisterTimn for code

**Comment by TomerFi** [comment]
@all-contributors please add @MisterTimn for code

**Comment by TomerFi** [comment]
@all-contributors please add @MisterTimn for code

**Comment by TomerFi** [comment]
Sorry for that - `all-contributors` was giving me a hard time.
I ended up using their CLI tool to do it manually: https://github.com/TomerFi/version-bumper/commit/379d420958d97f5f33aa0c3f3e12e75722b86772

**Comment by TomerFi** [comment]
@MisterTimn

The new [version 2.1.0](https://github.com/TomerFi/version-bumper/releases/tag/2.1.0) release includes your contribution.

Thank you!

**Pull request merged**
